### ISSUE TITLE
Rviz - recurring WARN msgs if "empty" link in robot URDF

### ISSUE THREAD

**Issue by JeremyZoss** [moveit/moveit_ros#214]
My robot models (URDFs) sometimes contain an "empty" link corresponding to the tool-flange center.  This link is specified with a fixed transform to the last arm link, but no geometry data.  The link is included in the manipulator chain because it's important to include this final transform in the manipulator kinematics.

However, when I add a moveit MotionPlanning plugin to rviz, rviz begins throwing warning messages related to the empty node.  Nominally, I get a warning message every time I move the robot (and the display is updated).  If I connect the "Planning Scene Topic" to a frequently-published topic (e.g. from a virtual robot state-publisher), the message is logged several times a second.

The message I get is:

```
[WARN] [timestamp]: Link [tool0] does not exist
ros-groovy-rviz/src/rviz/robot/robot.cpp:RobotLink* rviz::Robot::getLink:209
```

I'm logging the issue here, rather than in ros-visualization, because I only get this behavior when using the moveit plugins.  I don't see the same messages in arm_navigation, or when displaying a RobotModel rviz plugin.

Here's some brief troubleshooting I've done:
- The message is generated because:
  1. rviz::Robot::load() builds an internal list of rviz::RobotLinks based on the URDF model.
  2. "Invalid" links (with no geometry data) are not added to this list.
  3. When PlanningSceneDisplay updates its display, it scans through the list of links associated with the Planning Group and calls robot->getLink on the rviz::Robot representation.
  4. This call fails to find the "empty" link in its internal link list, and throws the warning message above.
- Adding any kind of trivial geometry (e.g. <sphere radius="0.000001"/>) to the "empty" link resolves this issue completely.
- The planning behavior works fine, other than the message-flooding.

**Comment by isucan** [comment]
This is very curious. I thought I had fixed this and I can't see what would cause the problem. Are you running with moveit_ros\* 0.3.25 or newer?

**Comment by JeremyZoss** [comment]
My ros-groovy-moveit-ros package is version 0.3.21 (20130305-1648), which looks to be the latest version available in the primary release repository.

Feel free to mark this fixed if you've already got it handled.  Just list what version you think it's fixed in (0.3.25?), so I can keep an eye out for it.

**Comment by isucan** [comment]
I believe it is fixed in 0.3.25, which will be out to public very soon.

**Comment by mikeferguson** [comment]
Yep, this is fixed by 0.3.25, which is currently on shadow-fixed (I have empty links as well).
